plugin/src/test/java/org/opensearch/ml/rest/RestMLRAGSearchProcessorIT.java

```diff
@@ -0,0 +1,589 @@
+        + "}";
+
+    private static final String BM25_SEARCH_REQUEST_WITH_CONVO_TEMPLATE = "{\n"
+        + "  \"_source\": [\"%s\"],\n"
+        + "  \"query\" : {\n"
+        + "    \"match\": {\"%s\": \"%s\"}\n"
+        + "  },\n"
+        + "   \"ext\": {\n"
+        + "      \"generative_qa_parameters\": {\n"
+        + "        \"llm_model\": \"%s\",\n"
+        + "        \"llm_question\": \"%s\",\n"
+        + "        \"memory_id\": \"%s\",\n"
+        + "        \"context_size\": %d,\n"
+        + "        \"message_size\": %d,\n"
+        + "        \"timeout\": %d\n"
+        + "      }\n"
+        + "  }\n"
+        + "}";
+
+    private static final String OPENAI_MODEL = "gpt-3.5-turbo";
+    private static final String BEDROCK_ANTHROPIC_CLAUDE = "bedrock/anthropic-claude";
+    private static final String TEST_DOC_PATH = "org/opensearch/ml/rest/test_data/";
+    private static Set<String> testDocs = Set.of("qa_doc1.json", "qa_doc2.json", "qa_doc3.json");
+    private static final String DEFAULT_USER_AGENT = "Kibana";
+    protected ClassLoader classLoader = RestMLRAGSearchProcessorIT.class.getClassLoader();
+    private static final String INDEX_NAME = "test";
+
+    // "client" gets initialized by the test framework at the instance level
+    // so we perform this per test case, not via @BeforeClass.
+    @Before
+    public void init() throws Exception {
+
+        /*
+        Response response = TestHelper
+            .makeRequest(
+                client(),
+                "PUT",
+                "_cluster/settings",
+                null,
+                "{\"persistent\":{\"plugins.ml_commons.rag_pipeline_feature_enabled\":true}}",
+                ImmutableList.of(new BasicHeader(HttpHeaders.USER_AGENT, DEFAULT_USER_AGENT))
+            );
+        assertEquals(200, response.getStatusLine().getStatusCode());
+        
+        response = TestHelper
+            .makeRequest(
+                client(),
+                "PUT",
+                "_cluster/settings",
+                null,
+                "{\"persistent\":{\"plugins.ml_commons.memory_feature_enabled\":true}}",
+                ImmutableList.of(new BasicHeader(HttpHeaders.USER_AGENT, DEFAULT_USER_AGENT))
+            );
+        assertEquals(200, response.getStatusLine().getStatusCode());
+        */
+
+        Response response = TestHelper
+            .makeRequest(
+                client(),
+                "PUT",
+                "_cluster/settings",
+                null,
+                "{\"persistent\":{\n"
+                    + "        \"plugins.ml_commons.trusted_connector_endpoints_regex\": [\n"
+                    + "            \"^https://runtime\\\\.sagemaker\\\\..*\\\\.amazonaws\\\\.com/.*$\",\n"
+                    + "            \"^https://api\\\\.openai\\\\.com/.*$\",\n"
+                    + "            \"^https://api\\\\.cohere\\\\.ai/.*$\",\n"
+                    + "            \"^http://127.0.0.1:1323/.*$\",\n"
```

Comment:
why do we need this?

Reply:
I must have used it for some local testing.  I will remove it.